test/interactions/dragInteractionTests.ts

```diff
@@ -0,0 +1,140 @@
+///<reference path="../testReference.ts" />
+
+var assert = chai.assert;
+
+describe("Interactions", () => {
+  describe("Drag", () => {
+    var SVG_WIDTH = 400;
+    var SVG_HEIGHT = 400;
+
+    var startPoint = {
+      x: SVG_WIDTH / 4,
+      y: SVG_HEIGHT / 4
+    };
+    var endPoint = {
+      x: SVG_WIDTH / 2,
+      y: SVG_HEIGHT / 2
+    };
+
+    var outsidePointPos = {
+      x: SVG_WIDTH * 1.5,
+      y: SVG_HEIGHT * 1.5
+    };
+    var constrainedPos = {
+      x: SVG_WIDTH,
+      y: SVG_HEIGHT
+    };
+    var outsidePointNeg = {
+      x: -SVG_WIDTH / 2,
+      y: -SVG_HEIGHT / 2
+    };
+    var constrainedNeg = {
+      x: 0,
+      y: 0
+    };
```

Comment:
If we have this constrain idea as a util, you could use the util instead of needing to define another var